### ISSUE TITLE
Invalid calling convention for arm64

### ISSUE THREAD

**Issue by sr-tream** [NationalSecurityAgency/ghidra#5350]
**Describe the bug**
Extra float variables stored to integer registers, instead of stack. Also stack align set to 4, instead of 8

**To Reproduce**
1. Open function with signature like `T(float1, float2, float3, float4, float5, float6, float7, int1, int2, int3, int4, int5, int6, float8, float9, int7, int8)`
2. Ghidra place this args in next stores: s0, s1, s2, s3, s4, s5, s6, s7, w0, w1, w2, w3, w4, w5, **w6**, **w7**, **sstack[0x0]:4**, **sstack[0x4]:4**

**Expected behavior**
Right stores fo this args: s0, s1, s2, s3, s4, s5, s6, s7, w0, w1, w2, w3, w4, w5, **sstack[0x0]:4**, **sstack[0x8]:4**, **w6**, **w7**

**Screenshots**
![image](https://github.com/NationalSecurityAgency/ghidra/assets/12231048/c0c446af-6548-4e65-9c85-9b2e0dc2fecf)

![image](https://github.com/NationalSecurityAgency/ghidra/assets/12231048/5760dea0-7001-499b-be34-59fd0159f164)

**Attachments**

Builtin `Specification extension`:
[__cdecl.xml.txt](https://github.com/NationalSecurityAgency/ghidra/files/11507482/__cdecl.xml.txt)

**Environment (please complete the following information):**
 - OS: ArchLinux
 - Java Version: 17
 - Ghidra Version: 10.4
 - Ghidra Origin: 033d2a2ec6b573a3b17f264d17f7c3d0cfb85f13

**Comment by caheckman** [comment]
This was addressed with 7c686cbfb7d80135657784dc5b8bdbfee5b02aa8

**Comment by sr-tream** [comment]
> This was addressed with [7c686cb](https://github.com/NationalSecurityAgency/ghidra/commit/7c686cbfb7d80135657784dc5b8bdbfee5b02aa8)

Yes, fixed. But now ghidra throw error, for function from issue, when enabled "Use Custom Storage":

```

2024-02-07 02:55:50 ERROR (AnalysisTaskWrapper) Analyzer Error: Analysis Task: Variadic Function Signature Override - For input string: "_" java.lang.NumberFormatException: For input string: "_"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:668)
	at java.base/java.lang.Integer.parseInt(Integer.java:786)
	at ghidra.program.model.data.DataTypeNameComparator.getConflictValue(DataTypeNameComparator.java:99)
	at ghidra.program.model.data.DataTypeNameComparator.compare(DataTypeNameComparator.java:82)
	at ghidra.program.model.data.DataTypeComparator.compare(DataTypeComparator.java:34)
	at ghidra.program.model.data.DataTypeComparator.compare(DataTypeComparator.java:25)
	at java.base/java.util.TimSort.mergeLo(TimSort.java:721)
	at java.base/java.util.TimSort.mergeAt(TimSort.java:518)
	at java.base/java.util.TimSort.mergeCollapse(TimSort.java:448)
	at java.base/java.util.TimSort.sort(TimSort.java:245)
	at java.base/java.util.Arrays.sort(Arrays.java:1307)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1721)
	at java.base/java.util.Collections.sort(Collections.java:179)
	at ghidra.program.database.data.DataTypeManagerDB.buildSortedDataTypeList(DataTypeManagerDB.java:964)
	at ghidra.program.database.data.DataTypeManagerDB.findDataTypes(DataTypeManagerDB.java:2060)
	at ghidra.program.database.data.DataTypeUtilities.findDataType(DataTypeUtilities.java:821)
	at ghidra.program.database.data.DataTypeUtilities.findExistingClassStruct(DataTypeUtilities.java:495)
	at ghidra.program.model.listing.VariableUtilities.findExistingClassStruct(VariableUtilities.java:875)
	at ghidra.program.model.listing.VariableUtilities.findOrCreateClassStruct(VariableUtilities.java:820)
	at ghidra.program.model.listing.VariableUtilities.findOrCreateClassStruct(VariableUtilities.java:854)
	at ghidra.program.model.listing.VariableUtilities.getAutoDataType(VariableUtilities.java:175)
	at ghidra.program.database.function.FunctionDB.updateParametersAndReturn(FunctionDB.java:846)
	at ghidra.program.database.function.FunctionDB.loadVariables(FunctionDB.java:725)
	at ghidra.program.database.function.FunctionDB.getParameterCount(FunctionDB.java:1213)
	at ghidra.program.database.function.FunctionDB.getParameterCount(FunctionDB.java:1211)
	at ghidra.app.plugin.core.string.variadic.FormatStringAnalyzer.usesVariadicFormatString(FormatStringAnalyzer.java:231)
	at ghidra.app.plugin.core.string.variadic.FormatStringAnalyzer.run(FormatStringAnalyzer.java:127)
	at ghidra.app.plugin.core.string.variadic.FormatStringAnalyzer.added(FormatStringAnalyzer.java:91)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:186)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:660)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:760)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:639)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:604)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:58)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:103)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:334)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

**Comment by ryanmkurtz** [comment]
@sr-tream Can you open a new issue for this new problem?
